verify-rpms/rpm_verifier.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+from typing import Callable
+
+import click
+
+
+def get_rpmdb(container_image: str) -> Path:
+    raise NotImplementedError()
+
+
+def get_rpms(rpmdb: Path) -> list[str]:
+    raise NotImplementedError()
+
+
+def get_unsigned_rpms(rpms: list[str], verifier: Callable[[str], bool]) -> list[str]:
+    return [rpm for rpm in rpms if verifier(rpm)]
+
+
+def is_rpm_unsigned(rpm: str) -> bool:
+    raise NotImplementedError()
+
+
+def generate_output(unsigned_rpms: dict[str, list[str]]) -> bool:
+    raise NotImplementedError()
+
+
+@click.command()
+@click.option(
+    "--container-image",
+    help="Reference to container image",
+    type=str,
+    multiple=True,
+)
+def main(container_image: list[str]):
+    rpms_dbs: dict[str, Path] = {img: get_rpmdb(img) for img in container_image}
```

Comment:
Since we extract RPM DBs as files into temporary directories, we need to account for deleting those directories.

I can see a few ways to handle this:
1. If every RPM DB is a single file, we can use the [TemporaryDirectory][td] stdlib class as a context manager to allocate a directory that gets deleted when we exit the context. We would need to pass the directory path into the code that extracts and uses each RMP DB.
2. If each RPM DB needs its own tmpdir, we can write a "get_rpm_dbs" context manager that would accept a list of images and pass the dictionary of image to RMPDB path onto the context block while deleting all of them.
3. As an alternative to option 2 - we can create some resource managing object that provides a function to allocate a tmpdir, works as a context manager and deletes all the directories when exiting the context.

[td]: https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Reply:
As far as I see it, this is currently planned to run on a short-lived container. Should we be concerned about this sort of cleanup at this stage?

Reply:
Each temp file takes memory. Its not significant with the current implementation, because we end up needing to allocate space (memory) for all the RPM DBs . But if we opt for threads witin thread pools of limited size, properly controlling temp files will allow for lower memory utilization at run time.

Reply:
ok. I think that if we take your advice to write the code for a single image and then call that code multiple times, then putting each such invocation inside its own context manager for temporary directory should be straightforward.